### PR TITLE
Expose metrics for Prometheus Operator

### PR DESCRIPTION
**Pull request by sbruzzese902**
Hello! I've exposed metrics using KEYCLOAK_STATISTICS environment variable, then I've created a ServiceMonitor to integrate with Prometheus Operator. I also added a namespace, I thought it was a good idea to segregate the instance.

Already tested locally, works fine.

### CODE REVIEW

File: katalog/keycloak/sm.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: keycloak-metrics
+  labels:
+    app: keycloak
```
Comment (lnovara) [inline katalog/keycloak/sm.yml:11]: There are already 2 other svcs with this label, I would change that to `keycloak-metrics` to ease the service discovery done through the Service Monitor.

File: katalog/keycloak/sm.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      port: metrics
+      scheme: http
+      path: /metrics
+  jobLabel: k8s-app
+  namespaceSelector:
+    matchNames:
+      - keycloak
+  selector:
+    matchLabels:
+      app: keycloak
```
Comment (lnovara) [inline katalog/keycloak/sm.yml:38]: There are already 2 other svcs with this label, I would change that to `keycloak-metrics` to ease the service discovery done through the Service Monitor.

File: katalog/keycloak/ns.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2022 SIGHUP s.r.l All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: keycloak
```
Comment (lnovara) [inline katalog/keycloak/ns.yml:8]: Minor nit: missing EOF

**Review by lnovara — CHANGES_REQUESTED** [review]
I am "worried" about the introduction of the namespace with respect to existing installation that already host Keycloak in an application namespace. In that scenario, the end user with have an empty namespace floating around for no good reason.

**Review by nutellinoit — CHANGES_REQUESTED** [review]
I agree with @lnovara

File: katalog/keycloak/sm.yml (hunk 1 above)
Comment (sbruzzese902) [reply katalog/keycloak/sm.yml:11]: Good point!
What about 2 labels, like `app: keycloak` and `component: metrics`?
Without the namespace I think the filter `app=keycloak` could be useful 😄

File: katalog/keycloak/ns.yml (hunk 3 above)
Comment (sbruzzese902) [reply katalog/keycloak/ns.yml]: Namespaced removed, as per your suggestion 😄

**Comment by ralgozino** [comment]
I agree with @lnovara about the service labels and the namespace.

Do we have a Grafana dashboard to see the metrics that we can add? If not, we should at least mention that the metrics are available in the readme IMHO.

**Review by ralgozino — CHANGES_REQUESTED** [review]
I agree with @lnovara about the service labels and the namespace.

Do we have a Grafana dashboard to see the metrics that we can add? If not, we should at least mention that the metrics are available in the readme IMHO.

File: katalog/keycloak/sm.yml (hunk 1 above)
Comment (lnovara) [reply katalog/keycloak/sm.yml:11]: I think it's fine, my point was to have a unique way to address the service associated to the metrics. 👍

File: README.md [hunk 4]
```diff
@@ -33,6 +33,10 @@ specific dependencies, please visit the single package's documentation:
 - :warning: Has issues
 - :x: Incompatible
 
+## Metrics
+
+The current package provides access to the metrics exposed by Keycloak, enabled using the env var `KEYCLOAK_STATISTICS=all`. Also, a ServiceMonitor is included, to be attached to [Prometheus Operator](https://github.com/sighupio/fury-kubernetes-monitoring/tree/master/katalog/prometheus-operator).
+
```
Comment (ralgozino) [inline README.md:39]: sorry for nitpicking, but I think it would be better to put this on the `katalog/keycloak/README.md` file instead

**Review by ralgozino — CHANGES_REQUESTED** [review]
> Honestly, I did not find suitable dashboards, but I don't think it's a problem, since the latest versions of Keycloak changed a lot (see [here](https://www.keycloak.org/migration/migrating-to-quarkus) if you're interested) and there will be dashboards when we will move to the new implementation as well. Also, there will be many more metrics available.

sounds good to me.

A minor comment on the readme update

**Review by ralgozino — APPROVED** [review]
thank you @sbruzzese902! 🚀

**Pull request merged**